src/scenes/level_scene.py

```diff
@@ -1971,6 +1971,49 @@ def new_turn(self) -> None:
             60,
         )
 
+    def leave_menu(self):
+        # This part I copied the code from function self.right_click
+        # to prevent being infected by future changes of function self.right_click
```

Comment:
I get your point, but what about code duplication?

Also, you for example fixed the cancelling of gold trades after cancelling the turn of a player, but since this code is not the same, the gold trade cancelling will happen only while cancelling with right click, not with left click/escape.
Code duplication should be avoid for this kind of situation: the different pieces of the project become harder to maintain on sync.

My suggestion: call this method inside of **right click** instead of having the same piece of code twice.
I will later try to refactor even more this piece of code because it's starting to get big, but that's another story.